src/test/java/com/ea/restaurant/service/impl/Oauth2ServiceImplTest.java

```diff
@@ -0,0 +1,100 @@
+package com.ea.restaurant.service.impl;
+
+import com.ea.restaurant.constants.Oauth2;
+import com.ea.restaurant.constants.Status;
+import com.ea.restaurant.entities.AppClient;
+import com.ea.restaurant.entities.AppClientScope;
+import com.ea.restaurant.entities.AppRefreshToken;
+import com.ea.restaurant.repository.AppAccessTokenRepository;
+import com.ea.restaurant.repository.AppClientRepository;
+import com.ea.restaurant.repository.AppClientScopeRepository;
+import com.ea.restaurant.repository.AppRefreshTokenRepository;
+import com.ea.restaurant.service.Oauth2Service;
+import com.ea.restaurant.utils.Oauth2Util;
+import com.ea.restaurant.utils.fixtures.AppClientFixture;
+import com.ea.restaurant.utils.fixtures.AppClientScopeFixture;
+import com.ea.restaurant.utils.fixtures.AppRefreshTokenFixture;
+import com.nimbusds.jose.JOSEException;
+import java.io.UnsupportedEncodingException;
+import java.util.Optional;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+public class Oauth2ServiceImplTest {
+  private Oauth2Service oauth2Service;
+  private AppClientRepository appClientRepository;
+  private AppClientScopeRepository appClientScopeRepository;
+  private AppAccessTokenRepository appAccessTokenRepository;
+  private AppRefreshTokenRepository appRefreshTokenRepository;
+  private String secretKey;
+  private AppClient appClient;
+  private AppClientScope appClientScope;
+  private AppRefreshToken appRefreshToken;
+
+  @BeforeEach
+  public void setUp() {
+    secretKey = "thisIsTheSecretKeyForTest-Etl001";
+    appClientRepository = Mockito.mock(AppClientRepository.class);
+    appClientScopeRepository = Mockito.mock(AppClientScopeRepository.class);
+    appAccessTokenRepository = Mockito.mock(AppAccessTokenRepository.class);
+    appRefreshTokenRepository = Mockito.mock(AppRefreshTokenRepository.class);
+    oauth2Service =
+        new Oauth2ServiceImpl(
+            secretKey,
+            appClientRepository,
+            appClientScopeRepository,
+            appAccessTokenRepository,
+            appRefreshTokenRepository);
+    MockitoAnnotations.openMocks(this);
+    appClient = AppClientFixture.buildAppClient(1L);
+    appClientScope = AppClientScopeFixture.buildAppClientScope(1L);
+    appRefreshToken = AppRefreshTokenFixture.buildAppRefreshToken(1L);
+  }
+
+  @Test
+  public void loginClient() throws JOSEException, UnsupportedEncodingException {
+    var mockedOauth2Util = Mockito.mockStatic(Oauth2Util.class);
+    Mockito.when(
+            Oauth2Util.buildClientCredentialsToken(
+                Mockito.eq(appClient),
+                Mockito.eq(appClientScope),
+                Mockito.eq(secretKey),
+                Mockito.eq(Oauth2.TokenType.ACCESS_TOKEN)))
+        .thenReturn("thisIsTheTestAccessToken");
```

Comment:
Looks like you use the sam string a lot, I think it is worth to move it to the app token fixture and re use it from there

Reply:
Done.